### PR TITLE
Skins :: separate play button / play_indicator + related improvements

### PR DESCRIPTION
**Pull request by ronso0**
This aims to prevent update issues with `play` / `play_indicator` buttons when using blinking Cue modes (see https://github.com/mixxxdj/mixxx/pull/2342#issuecomment-562957300 and below): the Play buttons would sometimes not update correctly because they had 3 connections attached (`play`, `play_indicator`, `cue_xx`)
Also they blinked like pressed / unpressed in LateNight which was confusing, or at least inconsistent.

Furthermore this unifies the Sampler play buttons.

- [x] Decks' **Play** button connections are now separated:
* top Layer is the actual button with `play`(left click) and `cue_set` (right click) connection
* bottom layer has the `play_indicator` connection, used to signal Cue mode blinking and previewing from Cue/hotcues
- [x] remove the `play_indicator` connection from samplers' **Play** button:
IMO it's distracting to have loaded samplers blink all the time with blinking Cue modes
- [x] set **Play** right-click in mini decks to `cue_default` in all skins to have a way to return to the Cue point. Overview hotcue icons, `play` and `cue_default` suffice to prepare a deck; for more detailed preparation we have the regular decks & waveforms
- [x] sampler Play buttons should all use a Pause icon when playing
- [x] add `play_cue_default` tooltip, use it for minimal decks' Play button

Idea:
* in Shade the sampler buttons in the deck row could use an indictator to signal the sampler is loaded (in other skins the sample title is an sufficient indicator)

### CODE REVIEW

**Comment by ronso0** [comment]
@Be-ing Could you please try to reproduce the indicator update hickups with this branch?

File: res/skins/Deere/deck_controls_row.xml [hunk 1]
```diff
@@ -52,16 +52,28 @@
...
-            <SetVariable name="ObjectName">PlayToggle</SetVariable>
-            <SetVariable name="MinimumSize"><Variable name="HorizontalStretchButtonMinimumSize"/></SetVariable>
-            <SetVariable name="MaximumSize"><Variable name="HorizontalStretchButtonMaximumSize"/></SetVariable>
-            <SetVariable name="SizePolicy"><Variable name="HorizontalStretchButtonSizePolicy"/></SetVariable>
-            <SetVariable name="left_connection_control"><Variable name="group"/>,play</SetVariable>
-            <SetVariable name="right_connection_control"><Variable name="group"/>,cue_set</SetVariable>
-            <SetVariable name="display_connection_control"><Variable name="group"/>,play_indicator</SetVariable>
-          </Template>
+          <WidgetGroup><!-- Play button + play_indicator -->
+            <Layout>stacked</Layout>
```
Comment (Be-ing) [inline res/skins/Deere/deck_controls_row.xml:56]: Hmm, this is an interesting hack but I suspect there is a bug in WPushButton or in the engine's handling of `play` or `play_handling` at the root of this...

Comment (ronso0) [reply res/skins/Deere/deck_controls_row.xml:56]: yeah, but maybe it's simply indicator confusion for buttons with two connections affecting the displayValue. now the Play button has only one connnection.

**Comment by Holzhaus** [comment]
This now conflicts with master.

**Comment by ronso0** [comment]
Conflicts resolved.

**Comment by Holzhaus** [comment]
@Be-ing Does this fix the issue for you? Shall we merge?

**Comment by daschuer** [comment]
It looks like this PR works around a widget bug. I am struggling to follow the discussion and understand what is the main issue fixed by the stacked layout. 

I like to dig for the root cause. 

Can you add a step by step guide, what to do to reproduce the issue, what is shown, 
what should be shown instead.

**Comment by ronso0** [comment]
I can't reproduce this anymore, just experinced this once.
@Be-ing  noticed this with Numark mode IIRC: deck paused, Cue & Play blink, press Play https://github.com/mixxxdj/mixxx/pull/2342#issuecomment-562957300

However, the changes proposed here solve some styling issues:
The blinking Play button (deck paused) would show the Pause icon in Deere & LateNight which is wrong and confusing. Also in LateNight the Play button would appear pressed when blinking, which is also wrong.

So I recommend to merge this regardless of the CO issue.

**Review by daschuer — APPROVED** [review]
LGTM, issue in Deere with toggling play and pause icon is fixed.

**Comment by daschuer** [comment]
@Be-ing Does this fix https://bugs.launchpad.net/mixxx/+bug/1856937 for you?

**Comment by daschuer** [comment]
Does this also fix https://bugs.launchpad.net/mixxx/+bug/1855620

**Pull request merged**